File: mmupnp/src/main/java/net/mm2d/upnp/empty/EmptyDevice.java
/*
 * Copyright (c) 2018 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.empty;

import net.mm2d.upnp.Action;
import net.mm2d.upnp.ControlPoint;
import net.mm2d.upnp.ControlPoints;
import net.mm2d.upnp.Device;
import net.mm2d.upnp.HttpClient;
import net.mm2d.upnp.Icon;
import net.mm2d.upnp.IconFilter;
import net.mm2d.upnp.Service;
import net.mm2d.upnp.SsdpMessage;

import java.util.Collections;
import java.util.List;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

public class EmptyDevice implements Device {
    @Override
    public void loadIconBinary(
            @Nonnull final HttpClient client,
            @Nonnull final IconFilter filter) {
    }

    @Nonnull
    @Override
    public ControlPoint getControlPoint() {
        return ControlPoints.emptyControlPoint();
    }

    @Override
    public void updateSsdpMessage(@Nonnull final SsdpMessage message) {
    }

    @Nonnull
    @Override
    public SsdpMessage getSsdpMessage() {
        return ControlPoints.emptySsdpMessage();
    }

    @Override
    public long getExpireTime() {
        return 0L;
    }

    @Nonnull
    @Override
    public String getDescription() {
        return "";
    }

    @Override
    public int getScopeId() {
        return 0;
    }

    @Nullable
    @Override
    public String getValue(@Nonnull final String name) {
        return null;
    }

    @Nullable
    @Override
    public String getValueWithNamespace(
            @Nonnull final String namespace,
            @Nonnull final String name) {
        return null;
    }

    @Nonnull
    @Override
    public String getLocation() {
        return "";
    }

    @Nonnull
    @Override
    public String getBaseUrl() {
        return "";
    }

    @Nonnull
    @Override
    public String getIpAddress() {
        return "";
    }

    @Nonnull
    @Override
    public String getUdn() {
        return "";
    }

    @Nullable
    @Override
    public String getUpc() {
        return null;
    }

    @Nonnull
    @Override
    public String getDeviceType() {
        return "";
    }

    @Nonnull
    @Override
    public String getFriendlyName() {
        return "";
    }

    @Nullable
    @Override
    public String getManufacture() {
        return null;
    }

    @Nullable
    @Override
    public String getManufactureUrl() {
        return null;
    }

    @Nonnull
    @Override
    public String getModelName() {
        return "";
    }

    @Nullable
    @Override
    public String getModelUrl() {
        return null;
    }

    @Nullable
    @Override
    public String getModelDescription() {
        return null;
    }

    @Nullable
    @Override
    public String getModelNumber() {
        return null;
    }

    @Nullable
    @Override
    public String getSerialNumber() {
        return null;
    }

    @Nullable
    @Override
    public String getPresentationUrl() {
        return null;
    }

    @Nonnull
    @Override
    public List<Icon> getIconList() {
        return Collections.emptyList();
    }

    @Nonnull
    @Override
    public List<Service> getServiceList() {
        return Collections.emptyList();
    }

    @Nullable
    @Override
    public Service findServiceById(@Nonnull final String id) {
        return null;
    }

    @Nullable
    @Override
    public Service findServiceByType(@Nonnull final String type) {
        return null;
    }

    @Nullable
    @Override
    public Action findAction(@Nonnull final String name) {
        return null;
    }

    @Override
    public boolean isEmbeddedDevice() {
        return false;
    }

    @Nullable
    @Override
    public Device getParent() {
        return null;
    }

    @Nonnull
    @Override
    public List<Device> getDeviceList() {
        return Collections.emptyList();
    }

    @Nullable
    @Override
    public Device findDeviceByType(@Nonnull final String deviceType) {
        return null;
    }

    @Nullable
    @Override
    public Device findDeviceByTypeRecursively(@Nonnull final String deviceType) {
        return null;
    }

    @Override
    public boolean isPinned() {
        return false;
    }
}

File: mmupnp/src/main/java/net/mm2d/upnp/empty/EmptyDevice.kt
/*
 * Copyright (c) 2018 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.empty

import net.mm2d.upnp.Action
import net.mm2d.upnp.ControlPoint
import net.mm2d.upnp.ControlPoints
import net.mm2d.upnp.Device
import net.mm2d.upnp.Icon
import net.mm2d.upnp.IconFilter
import net.mm2d.upnp.Service
import net.mm2d.upnp.SingleHttpClient
import net.mm2d.upnp.SsdpMessage

/**
 * Empty implementation of [Device].
 */
object EmptyDevice : Device {
    override val controlPoint: ControlPoint
        get() = ControlPoints.emptyControlPoint()
    override val ssdpMessage: SsdpMessage
        get() = ControlPoints.emptySsdpMessage()
    override val expireTime: Long
        get() = 0L
    override val description: String
        get() = ""
    override val scopeId: Int
        get() = 0
    override val location: String
        get() = ""
    override val baseUrl: String
        get() = ""
    override val ipAddress: String
        get() = ""
    override val udn: String
        get() = ""
    override val upc: String?
        get() = null
    override val deviceType: String
        get() = ""
    override val friendlyName: String
        get() = ""
    override val manufacture: String?
        get() = null
    override val manufactureUrl: String?
        get() = null
    override val modelName: String
        get() = ""
    override val modelUrl: String?
        get() = null
    override val modelDescription: String?
        get() = null
    override val modelNumber: String?
        get() = null
    override val serialNumber: String?
        get() = null
    override val presentationUrl: String?
        get() = null
    override val iconList: List<Icon>
        get() = emptyList()
    override val serviceList: List<Service>
        get() = emptyList()
    override val isEmbeddedDevice: Boolean
        get() = false
    override val parent: Device?
        get() = null
    override val deviceList: List<Device>
        get() = emptyList()
    override val isPinned: Boolean
        get() = false

    override fun loadIconBinary(client: SingleHttpClient, filter: IconFilter) = Unit
    override fun updateSsdpMessage(message: SsdpMessage) = Unit
    override fun getValue(name: String): String? = null
    override fun getValueWithNamespace(namespace: String, name: String): String? = null
    override fun findServiceById(id: String): Service? = null
    override fun findServiceByType(type: String): Service? = null
    override fun findAction(name: String): Action? = null
    override fun findDeviceByType(deviceType: String): Device? = null
    override fun findDeviceByTypeRecursively(deviceType: String): Device? = null
}
